src/components/PopupWithForm.js

```diff
@@ -5,6 +5,8 @@ export default class PopupWithForm extends Popup {
         super({popupSelector});
         this._popupForm = this._popupElement.querySelector(".modal__form");
         this._handleFormSubmit = handleFormSubmit;
+        this._submitButton = this._popupElement.querySelector(".modal__save-button");
+        this._submitButtonText = this._submitButton.textContent;
     }
 
```

Comment:
Please, delete `this.close();` from line 26 in `PopupWithForm`


* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

Reply:
Deleted and added in the blocks then after a successful response